store/src/test/scala/geotrellis/store/GeoTrellisPathSpec.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.store
+
+import org.scalatest._
+
+class GeoTrellisPathSpec extends FunSpec {
+  describe("GeoTrellisPathSpec") {
+
+    it("should fail to parse without a layer") {
+      val path = GeoTrellisPath.parseOption("file:///foo/bar")
+      assert(path == None)
+    }
+
+    it("should parse a local absolute file path without scheme") {
+      val path = GeoTrellisPath.parse("/absolute/path?layer=baz")
+      assert(path.value == "file:///absolute/path")
+      assert(path.layerName == "baz")
+    }
+
+    it("should parse a local absolute file path with scheme") {
+      val path = GeoTrellisPath.parse("file:///absolute/path?layer=baz")
+      assert(path.value == "file:///absolute/path")
+      assert(path.layerName == "baz")
+    }
+
+    it("should parse a local relative file path with scheme") {
+      val path = GeoTrellisPath.parse("file://relative/path?layer=baz")
+      assert(path.value == "file://relative/path")
+      assert(path.layerName == "baz")
+    }
+
+    it("should parse a local relative file path without scheme") {
+      val path = GeoTrellisPath.parse("relative/path?layer=baz")
```

Comment:
Lets try to make this behave as a URI with a `file` scheme? If nothing would fail that would be 💯

Reply:
👍 on it